brave/src/main/java/brave/sampler/RateLimitingSampler.java

```diff
@@ -0,0 +1,57 @@
+package brave.sampler;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Based on https://github.com/aws/aws-xray-sdk-java/blob/2.0.1/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/reservoir/Reservoir.java
+ *
+ * Uses only Java 6 APIs to be compatible with Brave Core requirements. Accounts for nanoTime()
+ * overflow, as unlikely as that might be.
+ */
+public class RateLimitingSampler extends Sampler {
+  public static Sampler create(int tracesPerSecond) {
+    if (tracesPerSecond <= 0) {
+      return Sampler.NEVER_SAMPLE;
+    }
+    return new RateLimitingSampler(tracesPerSecond);
+  }
+
+  private static long NANOS_PER_SECOND = 1_000_000_000;
+
+  private int tracesPerSecond;
+  private AtomicLong usage;
+  private AtomicLong nextUpdate;
+
+  RateLimitingSampler(int tracesPerSecond) {
+    this.tracesPerSecond = tracesPerSecond;
+    this.usage = new AtomicLong(0);
+    this.nextUpdate = new AtomicLong(getNextUpdateValue(System.nanoTime()));
+  }
+
+
+  @Override public boolean isSampled(long traceId) {
```

Comment:
Samll thing. The `traceId` is not used anywhere. Is it there for any future implementations for the WIP?

Reply:
It has to be there from the `Sampler` class we are extending. Maybe I could rename the variable to signal that a bit, but something has to be there

Reply:
if you notice the counting sampler ignores this. we can make a javadoc on the base type that says implementations that don't consider trace ID can ignore the input.

Reply:
I've added javadoc to the abstract method, and changed the variable name to signal that it is ignored